### PR TITLE
Stderr

### PR DESCRIPTION
**Pull request by stablestud**
Added some output which helps in debugging with an unittest.
Added connection retry (2x) if it fails. (`curl` and` wget` have now a timeout of 20sec)
Fixed scripts termination if one host was not reachable.

This is on build on [feedback](https://github.com/sedrubal/adaway-linux/pull/17) #17

### CODE REVIEW

File: adaway-linux.sh [hunk 1]
```diff
@@ -11,12 +11,11 @@
 
 # settings
 HOSTSORIG="/etc/.hosts.original"
-TMPDIR="/tmp/adaway-linux/"
+TMPDIR="/tmp/adaway-linux"
 #
 
 SCRIPT_DIR="$(cd "$(dirname "${0}")" && pwd)"  # Gets the location of the script
 
-set -e
```
Comment (sedrubal) [inline adaway-linux.sh]: Why did you remove this line?

File: adaway-linux.sh [hunk 2]
```diff
@@ -62,54 +61,60 @@ while read src; do
         # - remove additional localhost entries possibly picked up from sources
         # - remove remaining comments
         # - split all entries with one tab
-        if type curl 2>/dev/null > /dev/null ; then
-          DOWNLOAD_CMD=$(curl --progress-bar -L "${src}")
+        if type curl 1>/dev/null 2>&1; then
+          DOWNLOAD_CMD=$(curl --progress-bar -L --connect-timeout 20 --retry 2 "${src}")
         else
-          DOWNLOAD_CMD=$(wget "${src}" -nv --show-progress -L -O -)
+          DOWNLOAD_CMD=$(wget "${src}" -nv --show-progress --read-timeout=20 --timeout=20 -t 2 -L -O -)
```
Comment (sedrubal) [inline adaway-linux.sh]: Good idea. But we should add a variable `DL_TIMEOUT` and `DL_RETRIES` or something like that.

File: adaway-linux.sh [hunk 3]
```diff
@@ -51,8 +52,19 @@ mkdir -p "${TMPDIR}"
 
 # add domains from hosts-server listet in hostssources.lst
 while read src; do
+    timeout=20
+    retries=2
```
Comment (sedrubal) [inline adaway-linux.sh]: why did you not move this to the top and made it also `readonly`?

Comment (stablestud) [reply adaway-linux.sh]: I didn't considered the variables to be a global setting, this has been changed now to your preference. ;)
